moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -374,7 +374,11 @@ void PlanningScene::pushDiffs(const PlanningScenePtr& scene)
         scene->world_->removeObject(it.first);
         scene->removeObjectColor(it.first);
         scene->removeObjectType(it.first);
-        scene->getAllowedCollisionMatrixNonConst().removeEntry(it.first);
+        // if object is attached, it should not should not be removed from the ACM
+        if (!getCurrentState().hasAttachedBody(it.first))
```

Comment:
Sorry, this was wrong in my initial suggestion:
```suggestion
        if (!scene->getCurrentState().hasAttachedBody(it.first))
```